### PR TITLE
python3Packages.lark-parser: 0.11.2 -> 0.11.3, python3Packages.cassandra-driver: 3.24.0 -> 3.25.0, pythonPackages.gremlinpython: unbreak

### LINKED ISSUE #122350: python3Packages.gremlinpython does not build
**Describe the bug**
gremlinpython (a dependency of cassandra-driver, but otherwise no idea how important it is) fails to build.

**To Reproduce**
1. have a recent checkout of nixpkgs
2. nix-build -I nixpkgs=. -p python3Packages.gremlinpython
3. see how `setuptoolsBuildPhase` attempts to fetch something from the interwebs

**Notify maintainers**
@turion @ris

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.16, NixOS, 20.09.4058.22612485a46 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09.4058.22612485a46, nixos-unstable-21.05pre287333.63586475587"`
 - channels(cmm): `"home-manager-20.09, nixpkgs-20.09.4058.22612485a46, nixpkgs-unstable-21.05pre287333.63586475587"`
 - nixpkgs: `/home/cmm/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - python3Packages.gremlinpython
```

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change

* Update lark-parser to latest version
* unbreak downstream packages that came up while reviewing this:
	* Unbreak ``pythonPackages.gremlinpython``
	* Update ``python3Packages.cassandra-driver: 3.24.0 -> 3.25.0``.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121719` at 5be8755b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.hypothesmith</li>
  </ul>
</details>
<details>
  <summary>11 packages built successfully:</summary>
  <ul>
    <li>charliecloud</li>
    <li>python38Packages.cassandra-driver</li>
    <li>python38Packages.gremlinpython</li>
    <li>python38Packages.hypothesmith</li>
    <li>python38Packages.lark-parser</li>
    <li>python38Packages.radish-bdd</li>
    <li>python39Packages.cassandra-driver</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.lark-parser</li>
    <li>python39Packages.radish-bdd</li>
    <li>terraform-compliance</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `python`.
Near pkgs/development/python-modules/cassandra-driver/default.nix:1:53:
```
  |
1 | { stdenv, lib, buildPythonPackage, fetchFromGitHub, python, pythonOlder
  |                                                     ^
```
</li>
  </ul>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy for me nixos x86_64, LGTM.

Would be good to get a darwin reviewer though...

**Comment by drewrisinger** [comment]
Labeling as ZHF b/c of the fixed ``gremlinpython`` & ``cassandra-driver`` builds.
ZHF: #122042

File: pkgs/development/python-modules/gremlinpython/default.nix [hunk 1]
```diff
@@ -22,26 +32,44 @@ buildPythonPackage rec {
...
+      --replace 'importlib-metadata<3.0.0' 'importlib-metadata' \
+      --replace 'pytest-runner==5.2' 'pytest-runner'
   '';
 
-  nativeBuildInputs = [ pytestrunner ];  # simply to placate requirements
-  propagatedBuildInputs = [ six isodate tornado aenum ];
+  # setup-requires requirements
+  nativeBuildInputs = [
+    importlib-metadata
+    pytestrunner
```
Comment (dotlambda) [inline pkgs/development/python-modules/gremlinpython/default.nix:43]:
```suggestion
```

File: pkgs/development/python-modules/gremlinpython/default.nix [hunk 2]
```diff
@@ -22,26 +32,44 @@ buildPythonPackage rec {
       --replace 'PyHamcrest>=1.9.0,<2.0.0' 'PyHamcrest' \
       --replace 'radish-bdd==0.8.6' 'radish-bdd' \
       --replace 'mock>=3.0.5,<4.0.0' 'mock' \
-      --replace 'pytest>=4.6.4,<5.0.0' 'pytest'
+      --replace 'pytest>=4.6.4,<5.0.0' 'pytest' \
+      --replace 'importlib-metadata<3.0.0' 'importlib-metadata' \
+      --replace 'pytest-runner==5.2' 'pytest-runner'
```
Comment (dotlambda) [inline pkgs/development/python-modules/gremlinpython/default.nix:37]:
```suggestion
      --replace 'pytest-runner==5.2' ''
```

File: pkgs/development/python-modules/lark-parser/default.nix [hunk 3]
```diff
@@ -2,34 +2,36 @@
...
-  ];
-  disabledTests = [
-    "test_override_rule"  # has issue with file access paths
-  ];
+  checkPhase = ''
+    runHook preCheck
+
+    # Official way to run the tests. Runs unittest internally.
+    # pytest produces issues with some test resource paths (relies on __main__)
+    python -m tests
```
Comment (dotlambda) [inline pkgs/development/python-modules/lark-parser/default.nix:26]:
```suggestion
    ${python.interpreter} -m tests
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by rmcgibbo** [comment]
Here's what I got on darwin:

Result of `nixpkgs-review pr 121719` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.hypothesmith</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.cassandra-driver</li>
    <li>python39Packages.cassandra-driver</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python38Packages.gremlinpython</li>
    <li>python38Packages.hypothesmith</li>
    <li>python38Packages.lark-parser</li>
    <li>python38Packages.radish-bdd</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.lark-parser</li>
    <li>python39Packages.radish-bdd</li>
    <li>terraform-compliance</li>
  </ul>
</details>

**Comment by drewrisinger** [comment]
@rmcgibbo I don't know what the cassandra-driver failure on Darwin is about w/o the error log. Likely a test failure. I can selectively disable tests on Darwin to try to pass it, and let downstream people figure out if it's actually working or not.

I only unbroke gremlin & cassandra b/c they showed up in the ``nix-review``, no vested interest in these packages.

**Comment by dotlambda** [comment]
@ofborg build python3.pkgs.cassandra-driver

**Review by rmcgibbo — APPROVED** [review]

**Pull request merged**